File: packages/shared/src/__tests__/maxTripDays.test.ts
import { describe, it, expect } from 'vitest';
import { MAX_TRIP_DAYS } from '../types';

describe('MAX_TRIP_DAYS', () => {
  it('31 sabit (yaklaşık 1 ay)', () => {
    expect(MAX_TRIP_DAYS).toBe(31);
  });

  it('clamp davranışı: end > start+MAX-1 → kısıt', () => {
    // Welcome/Journey'de uygulanan clamp formülünün doğruluğu
    const start = '2026-10-01';
    const maxEnd = (() => {
      const d = new Date(start + 'T00:00:00Z');
      d.setUTCDate(d.getUTCDate() + MAX_TRIP_DAYS - 1);
      return d.toISOString().slice(0, 10);
    })();
    expect(maxEnd).toBe('2026-10-31');
  });
});

File: rotori-mobile/test/domain/max_trip_days_test.dart
// packages/shared/src/__tests__/maxTripDays.test.ts'in birebir Dart eşdeğeri.

import 'package:flutter_test/flutter_test.dart';
import 'package:rotori/domain/types.dart';

void main() {
  group('kMaxTripDays', () {
    test('31 sabit (yaklaşık 1 ay)', () {
      expect(kMaxTripDays, 31);
    });

    test('clamp davranışı: end > start+MAX-1 → kısıt', () {
      // Welcome/Journey'de uygulanan clamp formülünün doğruluğu
      final d =
          DateTime.utc(2026, 10, 1).add(const Duration(days: kMaxTripDays - 1));
      final maxEnd = d.toIso8601String().substring(0, 10);
      expect(maxEnd, '2026-10-31');
    });
  });
}
